google/resource_google_organization_iam_custom_role.go

```diff
@@ -118,12 +133,23 @@ func resourceGoogleOrganizationIamCustomRoleUpdate(d *schema.ResourceData, meta
 			if err := resourceGoogleOrganizationIamCustomRoleDelete(d, meta); err != nil {
 				return err
 			}
-		} else {
+			d.SetPartial("deleted")
+		}
+	}
+
+	// If role is not deleted, make sure it exists and undelete if needed.
```

Comment:
Because this gets called on create, the next bit about d.HasChange will actually return true and send a patch request. It's not a huge deal, but it's still an extra API call that we don't really need. If it's not hard, I think the Undelete should probably just get called straight from Create.

Reply:
I might be confused - isn't that what we want? If this gets called on create, what we want is the 1. role.undelete 2. role.patch. If we just call undelete then we still have to call patch to essentially reset the previously deleted role to whatever value we are trying to create.